src/trio/testing/_raises_group.py

```diff
@@ -35,12 +34,16 @@
     from typing import TypeVar
 
     MatchE = TypeVar("MatchE", bound=BaseException, covariant=True)
+
 # RaisesGroup doesn't work with a default.
 E = TypeVar("E", bound=BaseException, covariant=True)
-# These two typevars are special cased in sphinx config to workaround lookup bugs.
+E2 = TypeVar("E2", bound=BaseException)
+E3 = TypeVar("E3", bound=BaseException)
+Ec2 = TypeVar("Ec2", bound=Exception)
+Ec3 = TypeVar("Ec3", bound=Exception)
```

Comment:
Kinda weird to have `Ec2` and `Ec3` but no `Ec1`. Maybe `BaseExcT_co`, `BaseExcT_1`, `BaseExcT_2`, `ExcT_1`, `ExcT_2`?